mmpose/core/visualization/image.py

```diff
@@ -140,64 +140,73 @@ def imshow_keypoints(img,
-                        stickwidth = 2
-                        polygon = cv2.ellipse2Poly(
-                            (int(mX), int(mY)),
-                            (int(length / 2), int(stickwidth)), int(angle), 0,
-                            360, 1)
-                        cv2.fillConvexPoly(img_copy, polygon, color)
-                        transparency = max(
-                            0, min(1, 0.5 * (kpts[sk[0], 2] + kpts[sk[1], 2])))
-                        cv2.addWeighted(
-                            img_copy,
-                            transparency,
-                            img,
-                            1 - transparency,
-                            0,
-                            dst=img)
-                    else:
-                        cv2.line(img, pos1, pos2, color, thickness=thickness)
+                if pose_link_color[sk_id] is not None:
+                    pos1 = (int(kpts[sk[0], 0]), int(kpts[sk[0], 1]))
+                    pos2 = (int(kpts[sk[1], 0]), int(kpts[sk[1], 1]))
+                    if (pos1[0] > 0 and pos1[0] < img_w and pos1[1] > 0
+                            and pos1[1] < img_h and pos2[0] > 0
+                            and pos2[0] < img_w and pos2[1] > 0
+                            and pos2[1] < img_h
+                            and kpts[sk[0], 2] > kpt_score_thr
+                            and kpts[sk[1], 2] > kpt_score_thr):
+                        color = tuple(int(c) for c in pose_link_color[sk_id])
+                        if show_keypoint_weight:
+                            img_copy = img.copy()
+                            X = (pos1[0], pos2[0])
+                            Y = (pos1[1], pos2[1])
+                            mX = np.mean(X)
+                            mY = np.mean(Y)
+                            length = ((Y[0] - Y[1])**2 + (X[0] - X[1])**2)**0.5
+                            angle = math.degrees(
+                                math.atan2(Y[0] - Y[1], X[0] - X[1]))
+                            stickwidth = 2
+                            polygon = cv2.ellipse2Poly(
+                                (int(mX), int(mY)),
+                                (int(length / 2), int(stickwidth)), int(angle),
+                                0, 360, 1)
+                            cv2.fillConvexPoly(img_copy, polygon, color)
+                            transparency = max(
+                                0,
+                                min(1,
+                                    0.5 * (kpts[sk[0], 2] + kpts[sk[1], 2])))
+                            cv2.addWeighted(
+                                img_copy,
+                                transparency,
+                                img,
+                                1 - transparency,
+                                0,
+                                dst=img)
+                        else:
+                            cv2.line(
+                                img, pos1, pos2, color, thickness=thickness)
```

Comment:
```suggestion
                if (pos1[0] < 0 or pos1[0] >= img_w or pos1[1] < 0
                        or pos1[1] >= img_h or pos2[0] < 0 or pos2[0] >= img_w
                        or pos2[1] < 0 or pos2[1] >= img_h
                        or kpts[sk[0], 2] < kpt_score_thr
                        or kpts[sk[1], 2] < kpt_score_thr
                        or pose_link_color[sk_id] is None):
                    # skip the link that should not be drawn
                    continue
                color = tuple(int(c) for c in pose_link_color[sk_id])
                if show_keypoint_weight:
                    img_copy = img.copy()
                    X = (pos1[0], pos2[0])
                    Y = (pos1[1], pos2[1])
                    mX = np.mean(X)
                    mY = np.mean(Y)
                    length = ((Y[0] - Y[1])**2 + (X[0] - X[1])**2)**0.5
                    angle = math.degrees(math.atan2(Y[0] - Y[1], X[0] - X[1]))
                    stickwidth = 2
                    polygon = cv2.ellipse2Poly(
                        (int(mX), int(mY)), (int(length / 2), int(stickwidth)),
                        int(angle), 0, 360, 1)
                    cv2.fillConvexPoly(img_copy, polygon, color)
                    transparency = max(
                        0, min(1, 0.5 * (kpts[sk[0], 2] + kpts[sk[1], 2])))
                    cv2.addWeighted(
                        img_copy,
                        transparency,
                        img,
                        1 - transparency,
                        0,
                        dst=img)
                else:
                    cv2.line(img, pos1, pos2, color, thickness=thickness)
```
What about reducing the indent level to make the code more clear?

Reply:
makes sense.